### DISCUSSION TITLE
Response syntax for http response

### DISCUSSION THREAD

**Discussion by kaihendry** [KyleRoss/node-lambda-log#71] [Ideas]
Would be cool if there was a prescribed example to get a http response captured like so:
```
{
    "fields": {
        "app": "bus",
        "commit": "cc49127",
        "duration": 20,
        "ip": "154.205.14.233",
        "method": "GET",
        "path": "/",
        "plugin": "logs",
        "query": "id=11149",
        "region": "ap-southeast-1",
        "request_id": "78ec0f3a-b2d7-4e3a-9c55-2e74309572e7",
        "size": 1922,
        "stage": "production",
        "status": 200,
        "version": "144"
    },
    "level": "info",
    "timestamp": "2021-10-15T01:07:49.282652858Z",
    "message": "response"
}
```

For example how best to capture the ip?  x-forwarded-for?

The status is in a well known location, etc.

**Comment by KyleRoss** [comment]
Since every HTTP request module can be different in how this information is returned (unless you are using core Node), it would be hard to include an example. LambdaLog is a generic logger which will allow you to pass any information you wish into it.

With a typical Node response object (`http.IncomingMessage`), you'd get the client's IP address using:
```js
const ip = res.socket.remoteAddress;
```

and you can get the `X-Forwarded-For` header like:
```js
const forwardedFor = res.headers['x-forwarded-for'];
```

I hope that helps!

**Reply by kaihendry** [reply]
Hi Kyle, in the context of lambda, the IP is probably going to look like event.requestContext.identity.sourceIp right?

    log.options.meta.ip = event.requestContext.identity.sourceIp;

Sidenote, couldn't log lambda be initialised like so:

```
exports.lambdaHandler = lambdaLog((log) => async (event, context) => {
log.info("hello world");
})
```

**Reply by kaihendry** [reply]
I implemented the idea here https://github.com/kaihendry/lambda-test/blob/main/index.js#L27 @KyleRoss ... please tell me wdyt?

**Comment by kaihendry** [comment]
When the logger isn't global, it's quite the PITA to pass the logger around as an argument, right?

https://github.com/kaihendry/lambda-test/blob/ea3c7a55a3b8854aae742f468e92ae6008265f61/index.js#L33
